sup3r/preprocessing/samplers/cc.py

```diff
@@ -0,0 +1,179 @@
+            feature_sets=feature_sets,
+        )
+
+    def check_for_consistent_shapes(self):
+        """Make sure container shapes are compatible with enhancement
+        factors."""
+        enhanced_shape = (
+            self.lr_data.shape[0] * self.s_enhance,
+            self.lr_data.shape[1] * self.s_enhance,
+            self.lr_data.shape[2] * (1 if self.t_enhance == 1 else 24),
+        )
+        msg = (
+            f'hr_data.shape {self.hr_data.shape} and enhanced '
+            f'lr_data.shape {enhanced_shape} are not compatible with '
+            f'the given enhancement factors t_enhance = {self.t_enhance}, '
+            f's_enhance = {self.s_enhance}'
+        )
+        assert self.hr_data.shape[:3] == enhanced_shape, msg
+
+    def reduce_high_res_sub_daily(self, high_res, csr_ind=0):
+        """Take an hourly high-res observation and reduce the temporal axis
+        down to lr_sample_shape[2] * t_enhance time steps, using only daylight
+        hours on the middle part of the high res data.
+
+        Parameters
+        ----------
+        high_res : T_Array
+            5D array with dimensions (n_obs, spatial_1, spatial_2, temporal,
+            n_features) where temporal >= 24 (set by the data handler).
+        csr_ind : int
+            Feature index of clearsky_ratio. e.g. self.data[..., csr_ind] ->
+            cs_ratio
+
+        Returns
+        -------
+        high_res : T_Array
+            5D array with dimensions (n_obs, spatial_1, spatial_2, temporal,
+            n_features) where temporal has been reduced down to the integer
+            lr_sample_shape[2] * t_enhance. For example if hr_sample_shape[2]
+            is 9 and t_enhance = 8, 72 hourly time steps will be reduced to 9
+            using the center daylight 9 hours from the second day.
+
+        Note
+        ----
+        This only does something when `1 < t_enhance < 24.` If t_enhance = 24
+        there is no need for reduction since every daily time step will have 24
+        hourly time steps in the high_res batch data. Of course, if t_enhance =
+        1, we are running for a spatial only model so this routine is
+        unnecessary.
+
+        *Needs review from @grantbuster
+        """
+        if self.t_enhance not in (24, 1):
+            high_res = self.get_middle(high_res, self.hr_sample_shape)
+            high_res = nsrdb_reduce_daily_data(
+                high_res, self.hr_sample_shape[-1], csr_ind=csr_ind
+            )
+        return high_res
+
+    @staticmethod
+    def get_middle(high_res, sample_shape):
+        """Get middle chunk of high_res data that will then be reduced to day
+        time steps. This has n_time_steps = 24 if sample_shape[-1] <= 24
+        otherwise n_time_steps = sample_shape[-1]."""
+        n_days = int(high_res.shape[3] / 24)
+        if n_days > 1:
+            mid = int(np.ceil(high_res.shape[3] / 2))
+            start = mid - np.max((sample_shape[-1] // 2, 12))
```

Comment:
This all makes sense but we may want to revisit how the nighttime hours get padded. Right now the hourly observations during nighttime are garbage. Would be better to fix them at the nearest valid values. Does that make sense? If you plot up the hr values from this test you'll see what i mean (screenshot below). I cant remember what we did for the previous solarcc model but i dont think this will work well. 

https://github.com/NREL/sup3r/pull/220/files#diff-5c2bda4e2d2e1efcea8d269d4b3f953453e8ca2a59876b60c6e15ee9fbef14a3R46

Timestep around sunrise from the test: 
<img width="511" alt="image" src="https://github.com/user-attachments/assets/1ba99c43-1f48-4194-8027-25d9f29b508e">

Reply:
Yeah I thought about just limiting sample_shape[2] <= 24 for this. It seems that the previous behavior essentially defaulted to this, since the max size of the time dimension returned by reduce_high_res_sub_daily used to be 24. Would that accomplish what we want here?

Reply:
I actually think we're all good here. The sampling is working as designed and the reason for the weird patterns in the first image is just that the entire scene has GHI < 10 which leads to weird looking clearsky ratios. I added an issue with suggestions for ways to improve this but we don't need to do anything for this right now

https://github.com/NREL/sup3r/issues/229